dotnet-client-libraries/ITextBasics/Tests/UserMenuTest.cs

```diff
@@ -0,0 +1,32 @@
+﻿using ITextBasics;
+using ITextBasics.ConsoleManager;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+
+namespace Tests
+{
+    [TestClass]
+    public class UserMenuTest
+    {
+        [TestMethod]
+        public void GivenKey_WhenGettingUserInput_ThenExpectCorrectAction()
+        {
+            StringBuilder screenOutput = new();

```

Comment:
Since there is no checking of what has been printed to the console, and this is the only place where `FakeConsole` is being used. It seems like it would be better to remove this variable and then update `FakeConsole` to perform noops on the write and clear calls.
If there is an intended test case to test that UserMenu is writing the proper output to the console, then we should add a test around that as well.
In that case, we could also update `FakeConsole` to create its own internal buffer and add accessor methods around that. It would make the constructor cleaner and remove a dependency on callers, who would no longer have to care about how it was internally storing things. Maybe it could be using a list of strings, maybe a stringbuilder, maybe some kind of rolling array buffer, etc. The caller no longer has to care about that, it just calls the accessor methods to see what was written.

Reply:
Corrected class and add a test for a number of output lines on the console.